homeassistant/components/recorder/statistics.py

```diff
@@ -550,25 +550,26 @@ def compile_statistics(instance: Recorder, start: datetime) -> bool:
 
     _LOGGER.debug("Compiling statistics for %s-%s", start, end)
     platform_stats: list[StatisticResult] = []
+    old_metadata_dict: dict[str, tuple[int, StatisticMetaData]] = {}
     # Collect statistics from all platforms implementing support
     for domain, platform in instance.hass.data[DOMAIN].items():
         if not hasattr(platform, "compile_statistics"):
             continue
-        platform_stat = platform.compile_statistics(instance.hass, start, end)
+        platform_stat, metadata_dict = cast(
```

Comment:
Let's make it a data class so it's easier to read.

Reply:
83ebfb3913c9e2bce7ebea48dffc11cb8d41e702